plugins/inputs/logstash/logstash.go

```diff
@@ -0,0 +1,290 @@
+package logstash
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	jsonparser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+const sampleConfig = `
+  ## This plugin reads metrics exposed by Logstash Monitoring API.
+  # https://www.elastic.co/guide/en/logstash/current/monitoring.html
+  #
+  # logstashURL = "http://localhost:9600"
+  logstashURL = "http://localhost:9600"
+`
+const jvmStats = "/_node/stats/jvm"
+const processStats = "/_node/stats/process"
+const pipelineStats = "/_node/stats/pipeline"
+
+// Logstash Plugin Struct
+type Logstash struct {
+	LogstashURL string
+	client      *http.Client
+}
+
+// CommonData provides struct shared between all API calls
+type CommonData struct {
+	Host        string `json:"host"`
+	Version     string `json:"version"`
+	HTTPAddress string `json:"http_address"`
+	ID          string `json:"id"`
+	Name        string `json:"name"`
+}
+
+// JVMStats data structure
+type JVMStats struct {
+	CommonData CommonData
+	JVM        interface{} `json:"jvm"`
+}
+
+//ProcessStats data structure
+type ProcessStats struct {
+	CommonData CommonData
+	Process    interface{} `json:"process"`
+}
+
+//PluginEvents data structure
+type PluginEvents struct {
+	DurationInMillis float64 `json:"duration_in_millis"`
+	In               float64 `json:"in"`
+	Out              float64 `json:"out"`
+}
+
+//Plugin data structure
+type Plugin struct {
+	ID     string       `json:"id"`
+	Events PluginEvents `json:"events"`
+	Name   string       `json:"name"`
+}
+
+//PipelinePlugins data structure
+type PipelinePlugins struct {
+	Inputs  []Plugin `json:"inputs"`
+	Filters []Plugin `json:"filters"`
+	Outputs []Plugin `json:"outputs"`
+}
+
+//PipelineQueue data structure
+type PipelineQueue struct {
+	Events   float64     `json:"events"`
+	Qtype    string      `json:"type"`
+	Capacity interface{} `json:"capacity"`
+	Data     interface{} `json:"data"`
+}
+
+//Pipeline data structure
+type Pipeline struct {
+	Events  interface{}     `json:"events"`
+	Plugins PipelinePlugins `json:"plugins"`
+	Reloads interface{}     `json:"reloads"`
+	Queue   PipelineQueue   `json:"queue"`
+}
+
+//PipelineStats data structure
+type PipelineStats struct {
+	CommonData CommonData
+	Pipeline   Pipeline `json:"pipeline"`
+}
+
+//Description returns short info about plugin
+func (l *Logstash) Description() string { return "Read metricts exposed by Logstash" }
+
+//SampleConfig returns details how to configure plugin
+func (l *Logstash) SampleConfig() string { return sampleConfig }
+
+//createHttpClient create clients to access API
+func (l *Logstash) createHTTPClient() (*http.Client, error) {
+
+	tr := &http.Transport{
+		ResponseHeaderTimeout: time.Duration(3 * time.Second),
```

Comment:
This is probably following another example in the code, but I think we don't need a response header timeout.  The timeout on the http.Client should be enough.